### PR TITLE
EVSS Migration - 526 Submit - Transformation Service - basic framework

### PR DESCRIPTION
**Pull request by sethdarragile6**
## Summary

- Added the basic framework of a new transformation service that will ultimately convert EVSS-formatted JSON 526 requests
- *Disability Benefits Experience Team 1 (DBex TRex) 🦖
- No flipper needed, as it will not be called until further down the road

## Related issue(s)
- [61463](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/61463)

## Testing done

- Tests run successfully, they are simply validating the infrastructure (`bundle exec rspec spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb`)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/disability_compensation/requests/form526_request_body.rb
Comment (sethdarragile6) [inline lib/disability_compensation/requests/form526_request_body.rb:1]: Just note here that this file is especially long because it is a conversion from a very long Lighthouse submission request. It's very simple, with no business logic or anything- just structure. We also built a test file for this, but it was also very long and would have (far) exceeded the 500 line limit. This will be added in a separate PR after this PR has been merged in.

**Review by mchae-nava — APPROVED** [review]
LGTM! I do wonder if some form of automated contract testing would be worth considering, following the theme of covering as much as possible sans official testing solutions. Not that we have a great way to do that to begin with, but something to keep in the back

File: lib/disability_compensation/requests/form526_request_body.rb
Comment (ryan-mcneil) [reply lib/disability_compensation/requests/form526_request_body.rb:1]: Understood, but in the future, personally I'd rather see the specs in the same file and override LOC check. IMO it's better to have a spec alongside with the code it's testing. I think most approvers would agree.

**Review by ryan-mcneil — APPROVED** [review]
👍

File: lib/disability_compensation/requests/form526_request_body.rb
Comment (sethdarragile6) [reply lib/disability_compensation/requests/form526_request_body.rb:1]: Thanks, good to know for next time. We didn't think that the LOC check was overridable. 2nd PR coming soon.

**Pull request merged**
